astropy/units/core.py

```diff
@@ -2132,7 +2133,8 @@ def __call__(
                     warnings.warn(msg, UnitsWarning)
                 return UnrecognizedUnit(s)
 
-        elif isinstance(s, (int, float, np.floating, np.integer)):
+        # can't use Complex from .typing because of an import loop
```

Comment:
```suggestion
        # can't use Complex from units.typing because of an import loop
```

Reply:
If relative module paths are allowed in code then why shouldn't they be allowed in comments?

Reply:
well to tell the truth I'd like them to be forbidden everywhere 😅

Reply:
Given that there's multiple typing modules, including built in, disambiguating here is good. In general I would also tend to agree that absolute paths are to be preferred in comments, which aren't tested

Reply:
I should write down that importing `Complex` just before this `isinstance()` check still caused an import loop. I have ideas how to solve that, but that's for a different pull request.